core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSigner.java

```diff
@@ -112,6 +122,14 @@ private static boolean isEventStreaming(SdkHttpRequest request) {
         return "application/vnd.amazon.eventstream".equals(request.firstMatchingHeader("Content-Type").orElse(""));
     }
 
+    private static boolean isChunkedEncoding(SdkHttpRequest request) {
+        return "aws-chunked".equals(request.firstMatchingHeader(Header.TRANSFER_ENCODING).orElse(""));
+    }
+
+    private static boolean hasTrailer(SdkHttpRequest request) {
+        return request.firstMatchingHeader("x-amz-trailer").isPresent();
+    }
```

Comment:
This is likely temporary, just to exhibit the trailer decision. The actual choice will likely be determined by the presence of a flexible checksum or the parameters related to it.